PMM-7690 IAM roles (#579)
percona/pmm-doc: docs/setting-up/client/aws.md

![!image](../../_images/aws.iam.add-permissions.png)

## Creating an IAM role

> Alternatively to creating an IAM user you can also create an IAM role for a service to discover Amazon RDS DB instances automatically without the need for access and secret keys. This only works if you are running PMM through AWS.

Reviewer: ```suggestion
Instead of creating an IAM user you can create an IAM role for a service, to discover Amazon RDS DB instances automatically without the need for access and secret keys. (But this only works if you are running PMM through AWS.)
```

To create an IAM role open the IAM console and click *Roles* on the navigation pane.

1. Click the *Create role* button.

2. Select *AWS service* and select *EC2* for the use case.

3. Click the *Next: Permissions* button.

4. Find the policy created previously and select it.

5. Click the *Next: Tags* button.

6. (Optional) Add a metadata tag to the role.

7. Click the *Next: Review* button.

8. Fill the role name and description.

9. Click the *Create role* button

After the role is created EC2 instances running PMM will have permissions to discover RDS DB instances.

!!! note alert alert-primary ""

>     It’s also possible to create an IAM role to delegate permissions to an IAM user or to add permissions to a user belonging to other AWS account. Check the official AWS documentation for more information on  [Creating IAM roles](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create.html).

Reviewer: ```suggestion
    It’s also possible to create an IAM role to delegate permissions to an IAM user or to add permissions to a user belonging to another AWS account. See the [official AWS documentation on creating IAM roles](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create.html).
```
